xarray/core/indexing.py

```diff
@@ -722,20 +749,26 @@ def shape(self) -> _Shape:
         return np.broadcast(*self.key.tuple).shape
 
     def get_duck_array(self):
-        if isinstance(self.array, ExplicitlyIndexedNDArrayMixin):
-            array = apply_indexer(self.array, self.key)
-        else:
-            # If the array is not an ExplicitlyIndexedNDArrayMixin,
-            # it may wrap a BackendArray so use its __getitem__
+        from xarray.backends.common import BackendArray
+
+        if isinstance(self.array, BackendArray):
             array = self.array[self.key]
-        # self.array[self.key] is now a numpy array when
-        # self.array is a BackendArray subclass
-        # and self.key is BasicIndexer((slice(None, None, None),))
-        # so we need the explicit check for ExplicitlyIndexed
-        if isinstance(array, ExplicitlyIndexed):
+        else:
+            array = apply_indexer(self.array, self.key)
             array = array.get_duck_array()
         return _wrap_numpy_scalars(array)
 
+    async def async_get_duck_array(self):
+        print("inside LazilyVectorizedIndexedArray.async_get_duck_array")
+        from xarray.backends.common import BackendArray
+
+        if isinstance(self.array, BackendArray):
```

Comment:
I think this is a lot cleaner. In my previous refactor, I was trying hard to not depend on BackendArray but that's unavoidable now for async stuff AFAICT